src/base/static/components/input-form/input-form.js

```diff
@@ -0,0 +1,776 @@
+            mapConfig={mapConfig}
+            emitter={emitter}
+          />
+        ];
+        break;
+      case constants.BIG_TOGGLE_FIELD_TYPENAME:
+        const values = [fieldConfig.content[0].value, fieldConfig.content[1].value];
+        return [
+          fieldPrompt,
+          <BigToggleField
+            name={fieldConfig.name}
+            checked={fieldValues[fieldConfig.name] === fieldConfig.content[0].value}
+            labels={[fieldConfig.content[0].label, fieldConfig.content[1].label]}
+            values={[fieldConfig.content[0].value, fieldConfig.content[1].value]}
+            id={"input-form-" + fieldConfig.name}
+            onChange={evt => this.onToggleFieldChange(evt, [fieldConfig.content[0].value, fieldConfig.content[1].value])}
+          />
+        ];
+        break;
+      case constants.DROPDOWN_FIELD_TYPENAME:
+        return [
+          fieldPrompt,
+          <DropdownField
+            name={fieldConfig.name}
+            value={fieldValues[fieldConfig.name]}
+            options={fieldConfig.content}
+            onChange={evt => this.onFieldChange(evt, "value")}
+          />
+        ];
+        break;
+      case constants.DROPDOWN_AUTOCOMPLETE_FIELD_TYPENAME:
+        return [
+          fieldPrompt,
+          <AutocompleteComboboxField
+            name={fieldConfig.name}
+            options={fieldConfig.content}
+            placeholder={fieldConfig.placeholder}
+            id={"autocomplete-" + fieldConfig.name}
+            onChange={this.onValueNameFieldChange.bind(this)}
+            showAllValues={true}
+          />
+        ];
+        break;
+      case constants.COMMON_FORM_ELEMENT_TYPENAME:
+        const commonFormElementConfig = Object.assign(
+          {},
+          placeConfig.common_form_elements[fieldConfig.name],
+          {name: fieldConfig.name}
+        );
+        return this.buildField(commonFormElementConfig);
+        break;
+    }
+  }
+
+  validate() {
+    const { fieldValues, formValidationErrors } = this.state;
+    let newValidationErrors = [];
+
+    if (!this.geometry) {
+      newValidationErrors.push(messages.missingGeometry);
+    }
+
+    for (let i = 0; i < this.requiredFields.length; i++) {
+      if (!fieldValues[this.requiredFields[i]]) {
+        newValidationErrors.push(messages.missingRequired);
+        break;
+      }
+    }
+
+    if (newValidationErrors.length > 0) {
+      this.onInvalidSubmit(newValidationErrors);
+      return;
+    } else {
+      this.onValidSubmit();
+      return;
+    }
+  }
+
+  // due to https://stackoverflow.com/questions/8917921/cross-browser-javascript-not-jquery-scroll-to-top-animation
+  scrollTo(elt, to, duration) {
+    const difference = to - elt.scrollTop;
+    const perTick = difference / duration;
+    setTimeout(() => {
+      elt.scrollTop = elt.scrollTop + perTick;
+      if (elt.scrollTop === to) return;
+      this.scrollTo(elt, to, duration - 10);
+    }, 10);
+  }
+
+  onInvalidSubmit(errors) {
```

Comment:
nit: This method is so short, and it's only used once, is there a reason we can't put it inside the `validate` method?